ckanext/datastore/backend/postgres.py

```diff
@@ -1287,17 +1288,60 @@ def search_data(context, data_dict):
     _insert_links(data_dict, limit, offset)
 
     if data_dict.get('include_total', True):
-        count_sql_string = u'''SELECT count(*) FROM (
-            SELECT {distinct} {select}
-            FROM "{resource}" {ts_query} {where}) as t;'''.format(
-            distinct=distinct,
-            select=select_columns,
-            resource=resource_id,
-            ts_query=ts_query,
-            where=where_clause)
-        count_result = _execute_single_statement(
-            context, count_sql_string, where_values)
-        data_dict['total'] = count_result.fetchall()[0][0]
+        total_estimation_threshold = \
+            data_dict.get('total_estimation_threshold') or 0
+        estimated_total = None
+        if total_estimation_threshold != 0 and not (where_clause or distinct):
```

Comment:
I don't think 0 should be "off" in this case. How about we use the option to not pass this value when you want no limit. That lets sites apply their own default and validators to clamp this to a reasonable value like in #4484

Reply:
Agreed. I've changed it so that total_estimation_threshold=None means "off", which is also the default. And good idea - a site can clamp it by adding `limit_to_configured_maximum` to the schema (once #4484 is merged).